staging/src/k8s.io/client-go/examples/third-party-resources/main.go

```diff
@@ -78,19 +77,29 @@ func main() {
 	}
 
 	// make a new config for our extension's API group, using the first config as a baseline
-	var tprconfig *rest.Config
-	tprconfig = config
-	configureClient(tprconfig)
-
-	tprclient, err := rest.RESTClientFor(tprconfig)
+	exampleClient, exampleScheme, err := NewClient(config)
 	if err != nil {
 		panic(err)
 	}
 
+	// start a watcher on instances of our TPR
+	watcher := Watcher{
+		clientset:     clientset,
+		exampleClient: exampleClient,
+		exampleScheme: exampleScheme,
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	go watcher.Run(ctx)
+
+	time.Sleep(5 * time.Second) // Wait until watcher starts
```

Comment:
Which means that it is critical? I just want to avoid building cargo cults here because people copy this without understanding. So a proper 3 liner comment explaining this makes sense IMO.